warehouse/forklift/legacy.py

```diff
@@ -1354,6 +1262,40 @@ def file_upload(request):
                 },
             )
 
+        # If the user provided attestations, verify and store them
+        if "attestations" in request.POST:
+            attestation_service = request.find_service(
+                IAttestationsService, context=None
+            )
+
+            try:
+                attestations: list[Attestation] = attestation_service.parse(
+                    attestation_data=request.POST["attestations"],
+                    distribution=Distribution(
+                        name=filename, digest=file_hashes["sha256"]
+                    ),
+                )
+            except AttestationUploadError as e:
+                raise _exc_with_message(
+                    HTTPBadRequest,
+                    str(e),
+                )
+
+            attestations_db_models = attestation_service.persist_attestations(
+                attestations=attestations,
+                file=file_,
+            )
+
+            request.db.add_all(attestations_db_models)
+            generate_and_store_provenance_file(
+                request,
+                file_,
+                attestations,
+            )
```

Comment:
Any reason not to just make this part of `attestation_service.persist()`?

Reply:
Except that I need to find a better name to reflect that `persist_attestations` also stores provenance files, there is no good reason not to perform the call separately.

Of note, we could also merge `parse_attestations` and `persist_attestations`